moveit_core/trajectory_processing/src/time_optimal_trajectory_generation.cpp

```diff
@@ -962,22 +963,34 @@ bool TimeOptimalTrajectoryGeneration::computeTimeStamps(robot_trajectory::RobotT
   for (size_t p = 0; p < num_points; ++p)
   {
     moveit::core::RobotStatePtr waypoint = trajectory.getWayPointPtr(p);
-    Eigen::VectorXd new_point(num_joints);
+    Eigen::VectorXd new_point = Eigen::VectorXd::Zero(num_joints);
+
     // The first point should always be kept
-    bool diverse_point = (p == 0);
+    if (p == 0)
+    {
+      points.push_back(new_point);
+      continue;
+    }
```

Comment:
this is wrong, it adds a new zero vector whereas the original code adds the first vector, I have internal code that is broken by it. I would propose to revert all this.